### PR TITLE
saapby

### PR DESCRIPTION
**Pull request by saapby**

### CODE REVIEW

File: src/test/java/BrokenImageBMPTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import static helpers.Locators.get;
+
+/**
+ * Created by saap_by on 12.12.2015.
+ */
+public class BrokenImageBMPTest extends BmpTestBase {
+    private Har har;
+    private String url;
+    private By imageLink = get("brokenImage.links");
+    private String attribute = "src";
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageBMPTest.java]: this is not needed

File: src/test/java/BrokenImageBMPTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import static helpers.Helpers.getUrls;
+import static helpers.Helpers.mapToString;
+import static helpers.Locators.get;
+
+/**
+ * Created by saap_by on 12.12.2015.
+ */
+public class BrokenImageBMPTest extends BmpTestBase {
+    private Har har;
+    private String url;
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageBMPTest.java]: this can be local (better local)

File: src/test/java/BrokenImageBMPTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import static com.codeborne.selenide.Selenide.open;
+import static helpers.Helpers.getUrls;
+import static helpers.Helpers.mapToString;
+import static helpers.Locators.get;
+
+/**
+ * Created by saap_by on 12.12.2015.
+ */
+public class BrokenImageBMPTest extends BmpTestBase {
+    private Har har;
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageBMPTest.java]: Better make local

File: src/test/java/BrokenImageBMPTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    private String url;
+    private By imageLink = get("brokenImage.links");
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageBMPTest.java]: You should create a page object

File: src/test/java/BrokenImageBMPTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Test
+    public void brokenImageTest() {
+        har = server.getHar();
+        List<String> links = getUrls($$(imageLink), attribute);
+        Map<String, String> brokenImage = new HashMap<>();
+        for (HarEntry entry : har.getLog().getEntries()) {
+            url = entry.getRequest().getUrl();
+            int responseCode = entry.getResponse().getStatus();
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageBMPTest.java]: Let's make it with content type, it's not very good using just links, let's check all the images

**Comment by saapby** [comment]
corrected errors and I created second method finding broken image :)

File: src/main/java/pages/BrokenImageStaticPage.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package pages;
+
+import org.openqa.selenium.By;
+import static helpers.Locators.get;
+
+public class BrokenImageStaticPage {
+    public final static By IMAGE_LINK = get("brokenImage.links");
+    public final static String ATTRIBUTE = "src";
```
Comment (Kreisfahrer) [inline src/main/java/pages/BrokenImageStaticPage.java]: This is not needed, attributes can be hardcoded, they are constants

File: src/test/java/BrokenImageTest.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+public class BrokenImageTest extends TestBase {
+
+    @BeforeMethod
+    public void setup() {
+        open("broken_images");
+    }
+
+    @Test
+    public void simpleLoginTest() {
+        Map<String, String> brokenImage = new HashMap<>();
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageTest.java]: you can use List<String> or just String

File: src/test/java/BrokenImageTest.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        int imageSize = ($$(BrokenImageStaticPage.IMAGE_LINK)).size();
+        for (int i = 0; i < imageSize; i++) {
```
Comment (Kreisfahrer) [inline src/test/java/BrokenImageTest.java]: use foreach

**Comment by saapby** [comment]
I corrected errors

**Comment by saapby** [comment]
Class BmpTestBase.java uses field PRIVATE ProxyServer, but I use in my test PROTECTED ProxyServer
I think, that my version is the most correctly

File: src/main/java/core/BmpTestBase.java [hunk 9]
```diff
@@ -10,7 +10,8 @@
 import static org.openqa.selenium.net.PortProber.findFreePort;
 
 public class BmpTestBase extends TestBase{
-    private ProxyServer server;
```
Comment (saapby) [inline src/main/java/core/BmpTestBase.java]: private replace on protected

**Pull request merged**